### PR TITLE
New package: py-imbalanced-learn

### PR DESCRIPTION
**Pull request by meyersbs**

### CODE REVIEW

**Comment by meyersbs** [comment]
@spackbot fix style

**Comment by meyersbs** [comment]
@spackbot fix style

**Review by tldahlgren — COMMENTED** [review]
Confirmed the sha256.

File: var/spack/repos/builtin/packages/py-imbalanced-learn/package.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    variant("optional", default=False, description="Enable optional dependencies.")
+    depends_on("py-pandas@1.0.5:", when="+optional", type=("build", "run"))
+    depends_on("py-tensorflow@2.4.3:", when="+optional", type=("build", "run"))
+    depends_on("py-keras@2.4.3:", when="+optional", type=("build", "run"))
+
+    # From https://imbalanced-learn.org/stable/install.html#getting-started:
+    depends_on("py-joblib@1.1.1:", type=("build", "run"))
+    depends_on("py-threadpoolctl@2.0.0:", type=("build", "run"))
+    depends_on("py-cython@0.29.24:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-imbalanced-learn/package.py]:
```suggestion
    depends_on("py-cython@0.29.24:", type="build")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**